src/main/java/apoc/metrics/Metrics.java

```diff
@@ -0,0 +1,111 @@
+        public long t;
+        public double value;
+        public String metric;
+
+        public Neo4jMetricEntry(long t, double value, String metric) {
+            this.t = t;
+            this.value = value;
+            this.metric = metric;
+        }
+    }
+
+    public class Neo4jMetric {
+        public String name;
+        public long lastUpdated;
+        public String path;
+
+        public Neo4jMetric(String name, long lastUpdated, String path) {
+            this.name = name;
+            this.lastUpdated = lastUpdated;
+            this.path = path;
+        }
+    }
+
+    @Procedure
+    @Description("apoc.metrics.list() - get a list of available metrics")
+    public Stream<Neo4jMetric> list() {
+        Object metricsSetting = ApocConfiguration.allConfig.get("dbms.directories.metrics");
+
+        File metricsDir = new File(metricsSetting == null ?
+                ApocConfiguration.allConfig.get("unsupported.dbms.directories.neo4j_home") + File.separator + "metrics"
+                : metricsSetting.toString());
+
+        if (!metricsDir.exists() || !metricsDir.isDirectory()) {
+            return Stream.empty();
+        }
+
+        FileUtils.checkReadAllowed(metricsDir.toURI().toString());
+
+        return Arrays.asList(metricsDir.list())
+                .stream()
+                .filter(s -> s.endsWith(".csv"))
+                .map(metricFile -> {
+                    String metricName = metricFile.substring(0, metricFile.length() - 4);
+                    File f = new File(metricsDir + File.separator + metricFile);
+
+                    return new Neo4jMetric(metricName, f.lastModified(), f.getAbsolutePath());
+                });
+    }
+
+    @Procedure
+    @Description("apoc.metrics.get(metric_name) - retrieve a system metric")
+    /**
+     * This method is a specialization of apoc.load.csv, it just happens that the directory and file path
+     * are local.  For Neo4j metrics, see: https://neo4j.com/docs/operations-manual/current/monitoring/metrics/reference/
+     *
+     * The reason it's not suitable for users to use regular apoc.load.csv to get at the metrics file is that from the
+     * outside they have no way of knowing the home path where they're located, and apoc.load.csv doesn't allow
+     * relative paths.
+     */
+    public Stream<Neo4jMetricEntry> get(
+            @Name("metric_name") String metricName,
+            @Name(value = "config",defaultValue = "{}") Map<String, Object> config) {
+        // These config parameters are generally true of Neo4j metrics.
+        config.putIfAbsent("sep", ",");
+        config.putIfAbsent("header", true);
+
+        String neo4jHome = ApocConfiguration.allConfig.get("unsupported.dbms.directories.neo4j_home");
+        String url = "file://" + neo4jHome + File.separator + "metrics" + File.separator + metricName + ".csv";
+
+        FileUtils.checkReadAllowed(url);
+
+        return new LoadCsv().csv(url, config).map(entry -> {
```

Comment:
is apoc load csv overkill for this? because it does a lot of mapping and stuff, which might affect performance
compared to a plain CsvReader?

perahps something to consider for later.

Reply:
I don't think/hope it's too heavy -- my thought here was that I would need some options in some cases to read values back, and I wanted to reuse all of the config options that come with apoc.load.csv so they are not necessary to re-implement.

Reply:
Oki